src/library-authoring/LibraryAuthoringPage.tsx

```diff
@@ -205,6 +223,36 @@ const LibraryAuthoringPage = ({
     }
   }, [navigateTo]);
 
+  // Verify the migration task status
+  useEffect(() => {
+    if (migrationId) {
+      if (migrationStatusIsPending || migrationStatusIsError) {
+        return;
+      }
+
+      let deleteMigrationIdParam = false;
+      if (migrationStatusData.state === 'Succeeded') {
+        showToast(intl.formatMessage(migrationMessages.migrationSuccess));
+        queryClient.invalidateQueries({ predicate: (query) => libraryQueryPredicate(query, libraryId) });
+        deleteMigrationIdParam = true;
+      } else if (migrationStatusData.state === 'Failed') {
+        showToast(intl.formatMessage(migrationMessages.migrationFailed));
+        deleteMigrationIdParam = true;
+      } else if (migrationStatusData.state === 'Canceled') {
+        /* istanbul ignore next */
+        deleteMigrationIdParam = true;
+      }
+
+      if (deleteMigrationIdParam) {
+        params.delete('migration_task');
+        navigate({
+          pathname: location.pathname,
+          search: params.toString(),
+        });
```

Comment:
This will update the last history entry, so if you navigate back, you will be directed to the migration page instead of the library URL with the `?migration_task` parameter.
```suggestion
        }, { replace: true });
```

Reply:
Updated: https://github.com/openedx/frontend-app-authoring/pull/2493/commits/14ee1ab324db3a8d9dcd8f8c52da872f297d77ca